README.md

```diff
@@ -40,3 +40,26 @@ jobs:
           input: './annotations.json'
           title: 'Annotate Files'
 ```
+
+### Input file structure
+
+The below is the expected structure for annotations in TypeScript.
+
+```ts
+type AnnotationLevel = 'notice' | 'warning' | 'failure'
+
+interface Annotation {
+  message: string
+  path: string
+  line?: AnnotationSourcePositionReference
+  column?: AnnotationSourcePositionReference
+  level: AnnotationLevel
+}
+
+interface AnnotationSourcePositionReference {
+  start?: number
+  end?: number
+}
+
+type Annotations = Annotation[]
```

Comment:
don't like this.

Reply:
think `Annotation[]` is clear enough. `Annotations` just adds more cognitive load to readers.

Reply:
same here! I would assume it was an object

Reply:
In fact i can just reference the source; less overhead for maintenance